src/components/videos-you-might-like/index.js

```diff
@@ -15,20 +16,68 @@ const VideosYouMightLike = () => {
   const [videoPlayers, setVideoPlayers] = useState({});
   const [currentVideoId, setCurrentVideoId] = useState(null);
 
-  const onReady = function (videoId, event) {
+  const onReady = (videoId, event) => {
     setVideoPlayers({ ...videoPlayers, [videoId]: event.target });
   };
 
-  const stopAll = function () {
-    for (const videoPlayer of Object.values(videoPlayers)) {
-      videoPlayer.stopVideo();
-    }
+  const stopAll = () => {
+    document.querySelectorAll(".ytIFrame").forEach((iframe) => {
+      iframe.contentWindow.postMessage(
+        JSON.stringify({ event: "command", func: "stopVideo" }),
+        "*"
+      );
+    });
   };
 
-  const play = function (videoId) {
+  const play = (videoId) => {
     setCurrentVideoId(videoId);
     stopAll();
-    videoPlayers[videoId].playVideo();
+    videoPlayers[videoId]?.playVideo();
+  };
+
+  const youtubeVideos = () => {
+    return videos.map((video) => (
+      <YoutubeVideo
+        key={video.id}
+        videoId={video.id}
+        className={cx("h-[80%]", video.id !== currentVideoId && "hidden")}
+        onReady={onReady}
+      />
+    ));
+  };
+
+  const mainVideoPreview = () => {
```

Comment:
Is there a reason there are separate components rather than putting everything in the final `return`? (just curious)

Reply:
I thought the markup was becoming a bit too large to be readable so extracting the main parts (kinda like splitting a method into smaller ones) felt like a good idea, but I'm quite clueless about React/JSX's best practices.

Btw, how do you recommend looping and using conditions directly within JSX? The cleanest way I found was to use `{condition && <markup>}` (condition) and `{array.map((item) => <markup>}` (loops) but I'm curious what you recommend!

Reply:
If you feel that components are getting too big, you can cut them into smaller components. So for instance, right here `mainVideoPreview` could be its own component, in its own file:
```tsx
// src/components/videos-you-might-like/main-video-preview.js
export function MainVideoPreview() {
  return (
    …
  );
};
```

You could do the same with the other ones you've decoupled.

As for looping, an Array.map is typically what you want to do, yes. Just make sure to add a unique `key` attribute to each child of the loop. Conditional code can be done with `&&`, but it's a best practice to use ternary operators instead (see this [blog post](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx)):
```tsx
{condition && <h1>this is not preffered</h1>}
{condition ? <h1>but this is</h1> : null}
```

Reply:
Thanks! Yeah I did split some of it into their own components already (`YoutubeVideo` and `VideoPreview`) but I felt like those other ones would have been slightly overkill as they're very specific (basically just looping and including the above mentioned components by adding some CSS classes).